lib/ts/recipe/multifactorauth/types.ts

```diff
@@ -0,0 +1,150 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { BaseRequest, BaseResponse } from "../../framework";
+import OverrideableBuilder from "supertokens-js-override";
+import { GeneralErrorResponse } from "../../types";
+import RecipeUserId from "../../recipeUserId";
+import { SessionContainer } from "../session";
+import { SessionContainerInterface } from "../session/types";
+
+export type MFARequirement =
+    | {
+          id: string;
+          params?: any;
+      }
+    | string;
+
+export type MFARequirementList = (
+    | {
+          oneOf: MFARequirement[];
+      }
+    | {
+          allOf: MFARequirement[];
+      }
+    | MFARequirement
+)[];
+
+export type MFAClaimValue = {
+    c: Record<string, number>;
+    n: string[];
+};
+
+export type TypeInput = {
+    firstFactors?: string[];
+
+    getMFARequirementsForSession?: (
+        userId: string,
+        recipeUserId: RecipeUserId,
```

Comment:
- add an optional session object.
- add enabled factors for user.
- add completed factor for user. If a session does not exist, we get this from the recipe that is creating the session.

Reply:
Also check the last call recording to confirm if there is anything that is missed.